Prevent locale switcher in page chooser from breaking at the root level

gasman opened the pull request:
Fixes #8079

The bug occurs whenever the page chooser modal is opened from a page chooser widget where the initial value is the root page or an immediate child of it (as is the case when editing the initial site record). Not really up to speed on how multi-language is supposed to work, but here's my analysis:

The JS error was occurring because the switcher was missing [a toggle element](https://github.com/wagtail/wagtail/blob/096ab2a8b45d447ddb1ce8543df1af7703a9a00a/client/src/entrypoints/admin/core.js#L661) (i.e. one with a `data-dropdown-toggle` attribute). This is because it was not rendering the real locale selector widget from `wagtailadmin/shared/locale_selector.html`, but [the fake one in `wagtailadmin/shared/chooser_breadcrumb.html`](https://github.com/wagtail/wagtail/blob/096ab2a8b45d447ddb1ce8543df1af7703a9a00a/wagtail/admin/templates/wagtailadmin/shared/chooser_breadcrumb.html#L13-L18) that's used when `locale_options` is empty.

Looking at [where the page chooser view populates `locale_options`](https://github.com/wagtail/wagtail/blob/096ab2a8b45d447ddb1ce8543df1af7703a9a00a/wagtail/admin/views/chooser.py#L164-L215), there are two code paths: `parent_page_id is None` and `# We have a parent page (that is not the root page)`. However, `parent_page_id` is only None if the chooser modal has been launched from an initially blank chooser widget - this code overlooks the possibility that `parent_page_id` has been explicitly set to a top-level page. The fix therefore is to check if the resolved `parent_page` is a root page, rather than checking the incoming URL parameter.

The question remains whether the 'fake' locale switcher in chooser_breadcrumb.html has any legitimate use - if so, it should probably be fixed to include a `data-dropdown-toggle` element, and if not, it should be removed.

zerolab reviewed (approved):
Nice one @gasman

zerolab commented:
The fake switcher was there to provide a consistent style. it could probably be replaced by

```
<div aria-label="{{ page.locale.get_display_name }}" class="breadcrumb-link">
    {% icon name="site" class_name="default" %}
    {{ page.locale.get_display_name }}
</div>
```

gasman commented:
@zerolab Cheers, sounds good. I wasn't sure if there were any remaining cases where the fake switcher appears now that we've eliminated this one, but looking closer at the code I think it does come into play if the current locale is the only one available (i.e. at the top level, only one locale exists, or further down, the page has no translations) - so the HTML probably does still need tweaking to prevent the JS error in those cases.

zerolab commented:
>  I think it does come into play if the current locale is the only one available (i.e. at the top level, only one locale exists, or further down, the page has no translations)

That is the only case that I can think of too. You happy to tweak the HTML?

gasman commented:
Updated now:

* used @zerolab's HTML above for the 'fake' switcher used when only the current locale is available. (It still has a CSS hover style, which is 
* Removed the `{% if pages %}` condition from browse_results.html - it doesn't really serve any purpose, but meant that if you land on a parent page with no child pages (which is now possible with the locale switcher) the parent page was wrongly omitted too.

zerolab commented on wagtail/admin/templates/wagtailadmin/shared/chooser_breadcrumb.html:
```diff
@@ -10,11 +10,9 @@
-                        {% icon name="site" class_name="default" %}
-                        {{ page.locale.get_display_name }}
-                    </div>
+                <div aria-label="{{ page.locale.get_display_name }}" class="breadcrumb-link">
```
annoyingly this results in white on white if you start drilling down (e.g. home > blog)
the stylesheet caters for `.page-editor .modal .breadcrumb a` (giving it the body colour)

gasman replied:
Seems fine here (tried rebasing on current main in case something in the new designs has broken it) - am I missing something in the setup?
![Screenshot 2022-04-11 at 13 23 47](https://user-images.githubusercontent.com/85097/162738622-05618586-1e42-47de-a326-b98ae6ebb92e.png)

gasman commented:
Merged in 5473b5643f94a00ff6e3cbef0432de33c4d7fb50 (stable/2.16.x) and 47bf2910f3dcc563b0df1aa60d8caacd12e692a3 (main).
